frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx

```diff
@@ -0,0 +1,342 @@
+          );
+          const xTickFontSize = chartStyle.axes.ticks.fontSize;
+          const transform = isVertical
+            ? `rotate(-90, ${x} ${y}) translate(${Math.floor(
+                xTickFontSize * 0.9,
+              )} ${Math.floor(xTickFontSize / 3)})`
+            : undefined;
+
+          const textAnchor = isVertical ? "end" : "middle";
+
+          return {
+            ...props,
+            x,
+            y,
+            transform,
+            children: truncatedText,
+            textAnchor,
+          };
+        };
+
+  const numTicks = type === "timeseries" ? layout.numTicks : entries.length;
+  const tickLabelProps = getXTickLabelProps(chartStyle, isVertical);
+  const topMargin = settings.show_values
+    ? layout.margin.top + VALUES_MARGIN
+    : layout.margin.top;
+
+  const xTickHeight = isVertical ? xTickWidth : chartStyle.axes.ticks.fontSize;
+  const yTickWidth = getYTickWidth(
+    data,
+    accessors,
+    settings,
+    chartStyle.axes.ticks.fontSize,
+  );
+  const yLabelOffset = yTickWidth + layout.labelPadding;
+  const xMin = yLabelOffset + chartStyle.axes.labels.fontSize * 1.5;
+  const xMax = layout.width - layout.margin.right - layout.margin.left;
+  const xAxisHeight = getXAxisHeight(
+    xTickHeight,
+    measureTextHeight(chartStyle.axes.labels.fontSize),
+  );
+  let yMax = layout.height - xAxisHeight - topMargin;
+  let height = layout.height;
+  // If inner chart area is too short try to expand it but not more than MAX_EXTRA_HEIGHT
+  // to match what we do with XYChart (XYChart with legends can have be up to 340px tall)
+  if (yMax < MIN_INNER_HEIGHT) {
+    yMax =
+      minMax(
+        layout.height,
+        layout.height + MAX_EXTRA_HEIGHT,
+        yMax + xAxisHeight + MAX_EXTRA_HEIGHT,
+      ) - xAxisHeight;
+    height = topMargin + yMax + xAxisHeight;
+  }
+  const leftLabel = labels?.left;
+
+  const xScale = scaleBand({
+    domain: entries.map(entry => entry.x),
+    range: [0, xMax],
+    padding: layout.barPadding,
+  });
+
+  const yScale = scaleLinear({
+    domain: calculateWaterfallDomain(entries),
+    range: [yMax, 0],
+  });
+
+  const getBarProps = entry => {
+    const width = xScale.bandwidth();
+
+    const height = Math.abs(yScale(entry.start) - yScale(entry.end));
+    const x = xScale(entry.x);
+    const y = yScale(Math.max(entry.start, entry.end));
+
+    const fill = getWaterfallEntryColor(
+      entry,
+      getWaterfallColors(settings?.colors, getColor),
+    );
+
+    return { x, y, width, height, fill };
+  };
+
+  // Used only for rendering data point values
+  const series = calculateWaterfallSeriesForValues(
+    data,
+    accessors,
+    settings?.showTotal,
+  );
+
+  return (
+    <svg width={layout.width} height={height}>
+      <Group top={topMargin} left={xMin}>
+        <GridRows
+          scale={yScale}
+          width={xMax}
+          strokeDasharray={layout.strokeDasharray}
+        />
+      </Group>
```

Comment:
Move `GridRows` above `AxisBottom`, so the dash lines don't render on top of the X-axis line.